sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/models/ProgressReporter.java

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.callingserver.models;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+
+/**
+ * {@code ProgressReporter} offers a convenient way to add progress tracking to a given Flux.
+ */
+public final class ProgressReporter {
```

Comment:
Is there a timeline for when this needs to be GA'd? This functionality seems like it could become common and may want to be moved into azure-core.

FYI @JonathanGiles @srnagar

Reply:
I don't remember exact timeline for GA. Last time I knew it was about 45 days after Public Preview, so end of July if everything goes right.